pkg/providers/snow/snow.go

```diff
@@ -57,6 +57,7 @@ func NewProvider(kubeUnAuthClient KubeUnAuthClient, configManager *ConfigManager
 		retrier:          retrier,
 		configManager:    configManager,
 		skipIpCheck:      skipIpCheck,
+		bootstrapCreds:   &BootstrapCreds{},
```

Comment:
Yea based on the above comment, if we have a constructor here instead, would that help being able to avoid a combination of doing this and having a `Set` method?